internal/querynode/task.go

```diff
@@ -545,38 +525,48 @@ func (l *loadSegmentsTask) PreExecute(ctx context.Context) error {
 }
 
 func (l *loadSegmentsTask) Execute(ctx context.Context) error {
-	// TODO: support db
 	log.Info("LoadSegmentTask Execute start", zap.Int64("msgID", l.req.Base.MsgID))
 	err := l.node.loader.LoadSegment(l.req, segmentTypeSealed)
 	if err != nil {
-		log.Warn(err.Error())
+		for _, segment := range l.req.Infos {
+			l.node.metaReplica.removeSegment(segment.SegmentID, segmentTypeSealed)
```

Comment:
When loadSegment returns an error (for example, a segment already exists), the original segment will be deleted during cleanup.